File: internal/app/model/metric.go
package model

type PerformceUsage struct {
	Items   []PerformceUsageItem `json:"items"`
	CPUFree float64              `json:"cpuFree"`
	MemFree float64              `json:"memFree"`
}

type PerformceUsageItem struct {
	Name string  `json:"name"`
	CPU  float64 `json:"cpu"`
	Mem  float64 `json:"mem"`
}

type LogStatsticMetric struct {
	Executing int                     `json:"executing"`
	Items     []LogStatsticMetricItem `json:"items"`
}

type LogStatsticMetricItem struct {
	Count int64  `json:"count"`
	Date  string `json:"date"`
}

File: resources/src/api/metric.ts
import api from "./api";

export interface PerformceUsageItem {
  name: string;
  cpu: number;
  mem: number;
}

export interface PerformceUsage {
  items: PerformceUsageItem[];
  cpuFree: number;
  memFree: number;
}

export interface LogStatsticMetricItem {
  count: number;
  date: string;
}

export interface LogStatsticMetric {
  executing: number;
  items: LogStatsticMetricItem[];
}

export function getPerformceUsage() {
  return api.get<PerformceUsage>("/metric/performce").then((res) => res);
}

export function getLogMetric(dateType: number) {
  return api.get<LogStatsticMetric>("/metric/log", { dateType }).then((res) => res);
}

